### PR TITLE
Split up code into files, and cleaned output warnings

### PR DESCRIPTION
**Pull request by Diusrex**
One of a few pull request I will do to fix up the code. Will then work on updating the documentation.

### CODE REVIEW

**Comment by vahdat-ab** [comment]
@TimLethbridge I don't have the permission on this project  to merge this PR. You should do this.

**Comment by Nava2** [comment]
> @TimLethbridge I don't have the permission on this project to merge this PR. You should do this.

@vahdat-ab fixing..

File: src/DirectoryConverter.ump [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    
+    public void convertDirectory(Path saveTo) {
+        loadTemplates();
+        
+        saveTemplates(saveTo);
+    }
+    
+    private void loadTemplates() {
+        File dir = directory.toFile();
+          File[] directoryListing = dir.listFiles();
```
Comment (Nava2) [inline src/DirectoryConverter.ump]: spacing went wonky.

File: src/DirectoryConverter.ump [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+                      emittedTemplates.add(name);
+                  }
+                  
+                  // If it is a valid template file
+                  if (template.convert()) {
+                      templates.put(name, template);
+                  }
+            }
+        } else {
+            // TODO: Handle this case...
```
Comment (Nava2) [inline src/DirectoryConverter.ump]: Perhaps this should just throw an exception?

File: src/Elements.ump [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    }
+    
+    String text;
+}
+
+class ImportElement {
+    isA TemplateElement;
+    
+    public String getAsString(String mainClassToStore) {
+        // Should be changed to a format
```
Comment (Nava2) [inline src/Elements.ump]: Either a format or `StringBuilder`

File: src/Elements.ump [hunk 4]
```diff
@@ -0,0 +1,26 @@
+namespace cruise.umple.fromjet;
+
+interface TemplateElement {
+    String getAsString(String mainClassToStore);
+}
+
+class BasicElement {
```
Comment (Nava2) [inline src/Elements.ump]: Docs?

File: src/Elements.ump [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    isA TemplateElement;
+    
+    public String getAsString(String mainClassToStore) {
+        return text;
+    }
+    
+    String text;
```
Comment (Nava2) [inline src/Elements.ump]: Properties before methods, please.

File: src/TemplateConverter.ump [hunk 6]
```diff
@@ -0,0 +1,453 @@
...
+    }
+    
+    private String parseExpressionBlock(BufferedReader bufferedReader, String restOfLine) throws IOException {
+        StringBuffer buffer = new StringBuffer();
+        buffer.append("<<=");
+        
+           String line = restOfLine.substring("<%=".length());
+        
+        // Read all full lines first
+        while (!operationIsFinished(line, "%>")) {
```
Comment (Nava2) [inline src/TemplateConverter.ump]: This end tag feels like a "magic number," it should probably be a static field. There's a couple others that should be done, too.

File: src/TemplateConverter.ump [hunk 7]
```diff
@@ -0,0 +1,453 @@
...
+                    buffer.append(NewLine);
+                }
+            }
+            
+            // If there are only two lines, one ending on the previous guy, and they are separated by spaces, then need to add extra.
+                // Occurs in JavaClassGenerator, line 183
+            if (nextOperation == Operation.Code && previousLine != null) {
+                String generated = buffer.toString();
+                int numNewlines = (generated.length() - generated.replace(NewLine, "").length()) / NewLine.length();
+                if (containsOnlySpaces(previousLine) && (previousOp == Operation.Code || previousOp == Operation.Directive || previousOp == Operation.Expression) && numNewlines == 2) {
```
Comment (Nava2) [inline src/TemplateConverter.ump]: Lines of this length make it hard to read. :cry:

Comment (Nava2) [inline src/TemplateConverter.ump]: `previousOp == Operation.Code || previousOp == Operation.Directive || previousOp == Operation.Expression`
Could use an `EnumSet<Operation>` and just do `set.contains(previousOp)`. It's be nicer to read.

File: src/TemplateConverter.ump [hunk 8]
```diff
@@ -0,0 +1,453 @@
...
+        int indexOfStart = line.indexOf("<%");
+        if (indexOfStart != -1) {
+            return getStartingOperation(line.substring(indexOfStart));
+        }
+        
+        return Operation.Text;
+    }
+    
+    private Operation getStartingOperation(String line) {
+        if (line.startsWith("<%=")) {
```
Comment (Nava2) [inline src/TemplateConverter.ump]: You could attach this method to the `Operation` enum as a static method.

File: src/TemplateConverter.ump [hunk 9]
```diff
@@ -0,0 +1,453 @@
...
+    private String cutOffExtra(String line) {
+        // Means has whitespace followed by '<%@'
+        if (line != null && !shouldAppendNewline(line, Operation.Text)) {
+            line = line.substring(line.indexOf("<%@"));
+        }
+        
+        return line;
+    }
+    
+    private String removeAnyTabs(String line) {
```
Comment (Nava2) [inline src/TemplateConverter.ump]: Nice! Get that clean-up started. :+1:

**Comment by Nava2** [comment]
It looks great, @Diusrex. I'd like to see the "magic strings" disappear, though. The rest are nitpicks.

**Comment by Diusrex** [comment]
@Nava2 I completely agree, have a second pull request to change most of the magic strings and such! Was just going to do this one as a lead in, since the diffs wouldn't be very helpful if there is a mix of moving things around.

May just push it as well right now, just to resolve some of the issues.

**Comment by Nava2** [comment]
Sounds good, @Diusrex

**Pull request merged**